tests/test_unit/test_endpoint/test_modify_decorator.py

```diff
@@ -232,3 +232,77 @@ async def async_endpoint_error(
             )
             def get(self) -> int:
                 raise NotImplementedError
+
+
+def test_modify_with_set_cookie_in_headers_raises_error() -> None:
+    """@modify with Set-Cookie in headers= raise EndpointMetadataError."""
+    with pytest.raises(EndpointMetadataError, match='Set-Cookie'):
+
+        class _SetCookieHeaderController(Controller[PydanticSerializer]):
+            @modify(
+                status_code=HTTPStatus.OK,
+                headers={'Set-Cookie': NewHeader(value='session=abc123')},
+            )
+            def post(self) -> dict[str, str]:
+                return {'result': 'done'}  # pragma: no cover
+
+
+def test_modify_with_set_cookie_case_insensitive() -> None:
+    """@modify with Set-Cookie in headers= raise EndpointMetadataError."""
+    with pytest.raises(EndpointMetadataError, match='Set-Cookie'):
+
+        class _SetCookieHeaderController(Controller[PydanticSerializer]):
+            @modify(
+                status_code=HTTPStatus.OK,
+                headers={'set-cookie': NewHeader(value='session=abc123')},
```

Comment:
Just make the previous test parameterized on the header name. And do not duplicate the test.